### PR TITLE
Add ability to generate custom avatar

### PR DESCRIPTION
**Pull request by liPatrick**

### CODE REVIEW

File: app/api/generateCharacterImage/route.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        const imageBuffer = imageResponse.data; 
+
+        return new Response(imageBuffer, {
+            status: 200, 
+            headers: {
+                'Content-Type': imageResponse.headers['content-type'], // Set the appropriate content type for the image
+                'Content-Length': String(imageBuffer.length),
+            }
+        });
+
```
Comment (juberti) [inline app/api/generateCharacterImage/route.ts:26]: don't forget to `yarn format`

File: app/api/generateCharacterImage/route.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        return new Response(JSON.stringify({ error: e.message }), { status: 400 });
+    }
+}
+
+
+async function callAzureOpenAI(prompt: string) {
+    const endpoint = process.env.AZURE_OPENAI_ENDPOINT;
+    const key = process.env.AZURE_OPENAI_KEY;
+
+    const base_prompt = `Your job is to create festive avatars of North Pole characters, automatically rendering them in a cheerful, retro pixelated art style reminiscent of Super Nintendo 16-bit graphics. The GPT, with its concise and festive personality, focuses on the traditional holiday theme. It uses its creative judgment to fill in details, ensuring all avatars are in a pixelated style without requiring users to specify this. The goal is to provide users with delightful avatars that capture the essence of each character in a nostalgic, festive spirit.
```
Comment (juberti) [inline app/api/generateCharacterImage/route.ts:40]: This description should be rewritten to match a direct dall-e call, i.e., don't talk about a GPT, don't talk about responding with a question. I'd also suggest putting this as a constant at file scope for maximum readability.

File: app/api/generateCharacterImage/route.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    The avatar should have a plain background, and the image should feature the character's head only, never their shoulders or neck, or body. Feel free to include things like Santa hats, Poinsetta flowers, or other christmas and holiday accents. Also ensure the avatars use a 16-bit level of detail.
+    Never respond with a question, always respond with an image. Now, generate an image with the following description: `
+
+    const body = {
+        'model': 'dall-e-3',
+        'prompt': base_prompt+prompt,
```
Comment (juberti) [inline app/api/generateCharacterImage/route.ts:46]: should we add a newline after the base_prompt?

File: app/api/generateCharacterImage/route.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        const body = await req.json() as GenerateCharacterImageRequest;
+        if (typeof body !== 'object') {
+            throw new Error('Invalid request body: expecting object');
+        }
+        const characterDescription = body.characterDescription; // replace with actual data fields from your request
+        const openAIResponse = await callAzureOpenAI(characterDescription)
+        const imageUrl = openAIResponse.data[0].url; 
+
+        // Download the image 
+        const imageResponse = await axios.get(imageUrl, {responseType: 'arraybuffer'});
```
Comment (mdwelsh) [inline app/api/generateCharacterImage/route.ts:16]: I don't know that you need Axios for this - can you just use fetch?

File: app/api/generateCharacterImage/route.ts (hunk 2 above)
Comment (mdwelsh) [reply app/api/generateCharacterImage/route.ts:37]: Agreed - Maybe this goes in config.ts along with the other configuration constants?

File: app/components/ui/textarea.tsx [hunk 5]
```diff
@@ -8,7 +8,7 @@ const Textarea = React.forwardRef<HTMLTextAreaElement, TextareaProps>(({ classNa
   return (
     <textarea
       className={cn(
-        'flex min-h-[80px] w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50',
+        'flex min-h-[120px] w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50',
```
Comment (mdwelsh) [inline app/components/ui/textarea.tsx:11]: Please don't modify the base Textarea component. You can customize the specific use of Textarea by passing `className="min-h-[120px]"` to the instance you want; the whole idea with this `cn` utility is that it allows you to override defaults.

**Review by mdwelsh — CHANGES_REQUESTED** [review]

**Review by mdwelsh — APPROVED** [review]

**Pull request merged**